migrations/201906170001-Transaction-fixTaxAmount.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+import models, { sequelize, Op } from '../server/models';
+import Promise from 'bluebird';
+
+const fetchTransactions = async () => {
+  const query = { where: { taxAmount: { [Op.gt]: 0 } } };
+  if (process.env.LIMIT) {
+    query.limit = process.env.LIMIT;
+  }
+  return await models.Transaction.findAll(query);
```

Comment:
Can't do that otherwise I don't have all the info to re-create a new row in the Transactions table.

When migrations are run on a fresh db, they are run in order. So it should work just fine.

Reply:
You just have to put all the fields you need to re-create the new transaction in `attributes`. It won't work fine, we had the case multiple times in the past and had to fix it afterwards (ex: https://github.com/opencollective/opencollective-api/commit/e1d45155e5a8a41716747ad38896e6adcba6d92c).

Here is how:
1. You create a model like `Animal { name }`
2. You create a migration with `models.Animal.findAll(...)` in it.
3. You add a `type` column to Animal

When (2) is going to run for the first time (while (3) doesn't exist yet) your SQL query is going to look like this:

```sql
SELECT name FROM animal
```

No problem here. **But** when future contributors will run the migrations from scratch after (3) has been added to the codebase, the query from (2) will look like this:

```sql
SELECT name, type FROM animal
```

This is going to crash because `type` has not been created yet. 

The rule is: we shouldn't assume that the model and the database are synchronized when used in transactions (the goal of transactions being precisely to reconciliate the database with the model).

Reply:
Thanks for this thorough explanation! I learned something! :-)

Reply:
Not an easy one to think about, but seems logical once you've seen it :wink: